include/libhal/usb.hpp

```diff
@@ -17,6 +17,7 @@
 #include "functional.hpp"
 #include "scatter_span.hpp"
 #include "units.hpp"
+#include <optional>
```

Comment:
Follow libhal style guide for include order:

https://libhal.github.io/4.1/contributor_guide/style/